### ISSUE TITLE
Overhaul Android Editor UI with Floating Dialogs

### ISSUE THREAD

**Issue by syntaxerror247** [godotengine/godot-proposals#11196]
### Describe the project you are working on

Godot Android Editor.

### Describe the problem or limitation you are having in your project

The Android editor in Godot currently has a fixed UI layout, which doesn't adapt well to the limited screen space of Android devices. Panels like the Scene Tree, Inspector, and others are locked in place, leaving users with little flexibility to organize their workspace. This makes it challenging to work efficiently, especially when multiple panels are needed, as the layout can feel cramped and difficult to navigate.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The proposed enhancement is to make key panels in the Android editor, such as the Scene Tree and Inspector, into _floating_, movable panels with a _title bar_ to each panel for dragging, expanding, or collapsing. It would help make it more flexible and user friendly UI.

**1. Movable Dialogs:** Users can reposition dialogs anywhere on the screen, allowing them to prioritize important panels while keeping less used ones out of the way. This makes better use of available screen space, especially on small displays.

**2. Expandable/Collapsible Panels:** A title bar on each dialog would allow users to toggle between expanded (showing full content) and collapsed states (hiding the content but keeping the title visible). This minimizes clutter by letting users collapse panels they aren’t actively using.

**3. Persistent Layouts:** The editor would save the dialog positions and states, so users don’t need to rearrange them each time they restart the editor.

**4. Optional Docking:** Ofcourse, for those who prefer the current fixed layout, dialogs can be docked back into their original positions, ensuring flexibility for different workflows. But the default on Android editor would be _floating panels_.

### UI Behavior Overview

- Every Panel will includes a title, drag handle, and a expand/collapse button.
- The **title bar** will serves as a drag handle for repositioning the dialog.
- A button in the cornor of **title bar** will toggles the visibility of the content area.
- Dialog positions, sizes, and collapsed/expanded states are stored in the editor's settings.

### Mock-up

![Image](https://github.com/user-attachments/assets/4bebe692-57e9-43e5-bf94-9f34dd97dd23)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don’t think so. Please feel free to enlighten me if you know of a way. 

While the implementation may involve significant changes to the Android editor UI, it is in line with Godot's existing flexibility for desktop UI customizations.

### Is there a reason why this should be core and not an add-on in the asset library?

I don’t think this would be possible with an add-on and still provide the same level of polish. This feature should be core because it requires tight integration with the editor's UI and settings. This is a key improvement to the Android Editor UI, and it would make a significant difference for anyone using the Android editor.

**Comment by shahriarlabib000** [comment]
I think it should be added as an alternative UI rather than completely removing the original one, sure this could be set as the default UI for the editor. But the original UI should also be an option.

**Comment by syntaxerror247** [comment]
I don’t think we need to create an entirely new feature for the Android Editor. Since the Android Editor currently uses a "single window mode," the goal would be to make floating windows (currently only available in multi-window mode) work in single-window mode. This approach could also make the feature usable on other platforms. Personally, I find floating windows in the editor frustrating, as they hide behind the main editor once they lose focus.

**Comment by syntaxerror247** [comment]
- Related https://github.com/godotengine/godot-proposals/issues/8270

**Comment by m4gr3d** [comment]
> It looks like Android editor does supports and usages Multi-window mode by default.

@syntaxerror247 Multi-window is not supported yet on Android since that would require the use of multiple activities to properly support.

**Comment by syntaxerror247** [comment]
> > It looks like Android editor does supports and usages Multi-window mode by default.
> 
> @syntaxerror247 Multi-window is not supported yet on Android since that would require the use of multiple activities to properly support.

I thought the same, but then I noticed the Multi-Window setting was enabled in the editor settings. Maybe that's a bug.

That said, I think we can still implement the proposed feature using an embedded window, similar to how the Project Settings or Export windows work. We just need to turn off exclusive mode on `window` node, to let the input pass to parent node i.e. Editor.

**Comment by m4gr3d** [comment]
> That said, I think we can still implement the proposed feature using an embedded window, similar to how the Project Settings or Export windows work. We just need to turn off exclusive mode on window node, to let the input pass to parent node i.e. Editor.

Yes I agree! This approach can also be used on other platforms besides Android.

**Comment by Shadowblitz16** [comment]
I use a phone so can we get a extremely compact editor?

I noticed the android editor at anything greator then 1.8 scale gets cut off at the top.

And the buttons are too small. They could be taller for mobile but skinnier width wise with text scrolling.

Also using icons for tabs and having everything in a single dock space would be nice.
